packages/store/types/tests/store.lint.ts

```diff
@@ -137,4 +146,120 @@ describe('[TEST]: Action Types', () => {
     assertType(() => store.selectSnapshot({ foo: 'bar' })); // $ExpectError
     assertType(() => store.selectSnapshot()); // $ExpectError
   });
+
+  it('should be correct detect type with store methods in component', () => {
+    @State<string[]>({
+      name: 'todo',
+      defaults: []
+    })
+    class TodoState {
+      @Selector() // $ExpectType (state: string[]) => string[]
+      public static todo(state: string[]): string[] {
+        return state;
+      }
+    }
+
+    class TestComponent {
```

Comment:
I don't see the reason for wrapping these in a class called component. Am I mising something here?
They could just be assigned to variables, or even just used inline like the createSelector tests.

Reply:
done